olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Messages.h

```diff
@@ -0,0 +1,260 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <olp/dataservice/read/DataServiceReadApi.h>
+#include <olp/dataservice/read/model/Data.h>
+#include <olp/dataservice/read/model/StreamOffsets.h>
+#include <boost/optional.hpp>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace model {
+
+/**
+ * @brief Encapsulates information about actual data content.
+ */
+class DATASERVICE_READ_API Metadata final {
+ public:
+  /**
+   * @brief Gets the partition of this metata.
+   *
+   * @return The
+   * [partition](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/layers/partitions.html)
+   * to which this metadata content is related.
+   */
+  const std::string& GetPartition() const { return partition_; }
+  /**
+   * @brief Sets the partition ID of this metadata content.
+   *
+   * @param value The partition ID string. The maximum length is 500 characters.
+   */
+  void SetPartition(std::string value) { partition_ = std::move(value); }
+
+  /**
+   * @brief (Optional) Gets the checksum of this metadata.
+   *
+   * The algorithm used to calculate the checksum is user specific. It is not a
+   * secure hash. It is used only to detect changes in the content.
+   *
+   * Examples: `MD5` or `SHA1`
+   *
+   * @return The checksum of the content.
+   */
+  const boost::optional<std::string>& GetChecksum() const { return checksum_; }
+  /**
+   * @brief Sets the checksum of this metadata content.
+   *
+   * @see `GetChecksum` for information on the checksum.
+   *
+   * @param The checksum string.
+   */
+  void SetChecksum(boost::optional<std::string> value) {
+    checksum_ = std::move(value);
+  }
+
+  /**
+   * @brief (Optional) Gets the compressed size of the data in bytes.
+   *
+   * It is present only if the \ref `GetDataHandle` method is not empty.
+   * Applicable also if `Content-Encoding` is set to `gzip` when uploading and
+   * downloading data.
+   *
+   * @return The compressed size of content in bytes.
+   */
+  const boost::optional<int64_t>& GetCompressedDataSize() const {
+    return compressed_data_size_;
+  }
+  /**
+   * @brief Sets the compressed size of content.
+   *
+   * @see `GetCompressedDataSize` for information on the compressed size of
+   * content.
+   *
+   * @param value The compressed size of content in bytes.
+   */
+  void SetCompressedDataSize(int64_t value) { compressed_data_size_ = value; }
```

Comment:
Every optional member should also have a setter that accept optional paramater:

```suggestion
  void SetCompressedDataSize(boost::optional<int64_t> value) { compressed_data_size_ = value; }
```
Same for all other below.